odoo/addons/base/models/ir_cron.py

```diff
@@ -199,6 +199,17 @@ def _acquire_one_job(cls, cr, job_ids):
         # (ii) is implemented via the `WHERE` statement, when a job has
         # been processed, its nextcall is updated to a date in the
         # future and the optionnal trigger is removed.
+        #
+        # An `UPDATE` lock type is the strongest row lock, it conflicts
+        # with ALL other lock types. Among them the `KEY SHARE` row lock
+        # which is implicitely aquired by foreign keys to prevent the
+        # referenced record from being removed while in use. Because we
+        # never delete acquired cron jobs, foreign keys are safe to
```

Comment:
Never delete *or update FK targets* (which is basically always the `id` field, which we don't touch after creation I think).